src/catalog/database_catalog.cpp

```diff
@@ -977,7 +986,8 @@ std::vector<index_oid_t> DatabaseCatalog::GetIndexOids(
   auto *key_pr = oid_pri.InitializeRow(buffer);
   *(reinterpret_cast<table_oid_t *>(key_pr->AccessForceNotNull(0))) = table;
   std::vector<storage::TupleSlot> index_scan_results;
-  indexes_table_index_->ScanKey(*txn, *key_pr, &index_scan_results);
+  transaction::TransactionContext fake_txn(transaction::timestamp_t{UINT64_MAX}, transaction::timestamp_t{UINT64_MAX});
```

Comment:
why do you use a fake txn here with max timestamp instead of txn? Is this safe?

Reply:
This will also be communicated in the design doc. Essentially, we want to use a fake txn in order to get any indices that are logically invisible to us (e.g. whose commit timestamp is after our start timestamp) so that we can still insert into them and keep them up-to-date.

Is this safe? Yes, from an MVCC viewpoint indices are performance hints anyways and are not going to affect the result. From a logical viewpoint we're only ever reading from live indices which are guaranteed to be correct (even if the commit timestamp is later than our sstart timestamp)

Reply:
Alive is not necessarily congruent with visible - e.g. a transaction may be alive and still invisible to another transaction, so you would still need to do this

Reply:
> I think you can define the fake_txn_timestamp in some file like txn_def and document it.

We considered this, but weren't sure where best to put it so we put it off. We'll take another look at it